src/app/ui/question/question.component.ts

```diff
@@ -68,4 +70,8 @@ export class QuestionComponent implements OnInit {
   arr_diff(a1, a2) {
     return a1.filter(e => !a2.includes(e));
   }
+
+  sendRate() {
+    this.db.addSolvedQuestion(this.userId, {linkedQuestionId: this.qId, difficulty: this.selected}).then(() => {});
```

Comment:
You are ignoring the promise. There is no meaning to empty 'then'.
Please add NgxSpinner while waiting for the operation to complete, and show a toast after that.

Reply:
Solved.